protocol/consistencychecks.go

```diff
@@ -62,83 +61,82 @@ func NewCC(savedSTR []byte, useTBs bool, signKey sign.PublicKey) *ConsistencyChe
 	return cc
 }
 
-// Verify verifies the directory's response for a request.
-// It first verifies the directory's returned status code of the request.
+// clone makes a copy of the current consistency state.
+func (cc *ConsistencyChecks) clone() *ConsistencyChecks {
+	newCC := &ConsistencyChecks{
+		CurrentEpoch: cc.CurrentEpoch,
+		SavedSTR:     cc.SavedSTR,
+		useTBs:       cc.useTBs,
+		signKey:      cc.signKey,
+	}
+
+	if cc.useTBs {
+		newCC.TBs = make(map[string]*m.TemporaryBinding)
+		for k, v := range cc.TBs {
+			newCC.TBs[k] = v
+		}
+	}
+
+	return newCC
+}
+
+func (cc *ConsistencyChecks) update(updated *ConsistencyChecks) {
+	cc.CurrentEpoch = updated.CurrentEpoch
+	cc.SavedSTR = updated.SavedSTR
+	cc.TBs = updated.TBs
+}
+
+// Verify verifies the directory's response for a request and
+// updates the current state if all the verifications are successful.
+// It makes a copy of the current state and does verifications
```

Comment:
Maybe I'm misunderstanding something, but I don't understand how the client can be verifying a cached TB during a registration, when we discussed that client's [shouldn't be actively verifying](https://github.com/coniks-sys/coniks-go/pull/74#discussion_r84589208) all cached TBs (or did we decide against this?)

I've responded to [this](https://github.com/coniks-sys/coniks-go/pull/74#discussion_r85864068) comment separately.